runtime/src/asset.rs

```diff
@@ -89,13 +90,48 @@ pub trait Trait:
     type Currency: Currency<Self::AccountId>;
 }
 
+/// The type of an asset represented by a token.
+#[derive(codec::Encode, codec::Decode, Clone, PartialEq, Debug)]
+pub enum AssetType {
+    Equity,
+    Debt,
+    Commodity,
+    StructuredProduct,
+    Custom(Vec<u8>),
+}
+
+impl Default for AssetType {
+    fn default() -> Self {
+        AssetType::Custom(b"undefined".to_vec())
+    }
+}
+
+/// The type of an identifier associated with a token.
+#[derive(codec::Encode, codec::Decode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum IdentifierType {
+    Isin,
```

Comment:
I think it will be better if we store the identifier number along with type. i.e `Isin` -> `Isin(Vec<u>)`. We'll then be able to store them like `<IdentifierType>` rather than `<IdentifierType, Vec<u8>>`.

Reply:
The idea behind `IdentifierType` is that we can store multiple identifiers of different types in the same token. If we store the value in the enum variant, we won't be able to differentiate by type in the map.